### PR TITLE
Bump z2jh closer to v2.0

### PR DESCRIPTION
**Pull request by yuvipanda**
I've been running z2jh close to master at Berkeley for a few weeks without issue
(see https://github.com/berkeley-dsep-infra/datahub/issues/3418). This PR bumps 
us to latest master of z2jh 2.0. 

This will help with:

- https://github.com/2i2c-org/infrastructure/issues/1253
- https://github.com/2i2c-org/infrastructure/issues/1004

# Upgrading jupyterhub package in our user images

The pangeo images already use jupyterhub >= 2.3.x in their images,
and it's working on with our current version of z2jh. This PR will help match
those. 

Testing across our other hubs, everything seems to work fine with older
versions of jupyterhub in the image too - at least with jupyterhub 1.5.x.
With 1.3.x, the default URL doesn't take effect. I'll open separate PRs to 
bump up versions of jupyterhub:

- https://github.com/2i2c-org/2i2c-hubs-image/pull/1
- https://github.com/2i2c-org/utoronto-image/pull/31

Refs
* https://github.com/2i2c-org/infrastructure/issues/233
* https://github.com/2i2c-org/infrastructure/issues/1055

### CODE REVIEW

**Review by sgibson91 — APPROVED** [review]

**Comment by yuvipanda** [comment]
This was blocked on the meom-ige hub using a user image with too old a JupyterHub. They're testing a newer pangeo-notebook version on staging to see if that works, and then we can unblock this.

**Pull request merged**
